packages/cli/src/lib/setup/setupInstall.ts

```diff
@@ -385,14 +405,50 @@ export class Install {
                 console.error(`host.${hostname} Cannot install ${npmUrl}: ${result.exitCode}`);
                 return this.processExit(EXIT_CODES.CANNOT_INSTALL_NPM_PACKET);
             }
-            // create file that indicates, that npm was called there
+            // create file that indicates that npm was called there
             fs.writeFileSync(path.join(installDir, 'iob_npm.done'), ' ');
             // command succeeded
             return { _url: npmUrl, installDir: path.dirname(installDir) };
         } else {
-            console.error(result.stderr);
-            console.error(`host.${hostname} Cannot install ${npmUrl}: ${result.exitCode}`);
-            return this.processExit(EXIT_CODES.CANNOT_INSTALL_NPM_PACKET);
+            // if we see ENOTEMPTY error
+            if (!isRetry && result.stderr.includes('ENOTEMPTY')) {
+                // try to manually remove the directory and start installation again
+                // detect node_modules folder
+                const pack = await tools.detectPackageManagerWithFallback();
+                // extract adapter name
+                const name = npmUrl
+                    .replace(/\\/g, '/')
+                    .replace(/\.git$/, '')
+                    .split('/')
+                    .pop();
+                
+                if (name) {
+                    const folderPath = path.join(pack.cwd, 'node_modules', name.toLowerCase());
+                    console.error(
+                        `host.${hostname} Cannot install ${npmUrl}: try to delete adapter folder manually ("${folderPath}")`
+                    );
+                    let success = false;
+                    // delete adapter folder in node_modules
+                    try {
+                        deleteFoldersRecursive(folderPath);
```

Comment:
We know the "usual pattern" of the problematic directorynames for that "error 25" issue and we should try to validate that the pattern is that before deleting. There might be cases where it is something else and then we would start to delete stuff ... that might be not good

Reply:
Done